plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/RefactoringService.java

```diff
@@ -257,53 +258,128 @@ public RenameRefactoringSession createRenameRefactoring(CreateRenameRefactoring
         elementToRename = null;
     }
     if (elementToRename == null) {
-      throw new RefactoringException("Can't find java element to rename.");
+      throw new RefactoringException("Can't find java element to recommend.");
     }
 
+    JavaRenameRecommend.javaElement = elementToRename;
+
     return manager.createRenameRefactoring(
         elementToRename, cu, settings.getOffset(), settings.isRefactorLightweight());
   }
 
   /**
-   * Apply linked mode rename refactoring.
+   * Apply linked mode recommend refactoring.
    *
    * @param refactoringApply linked mode setting and refactoring session id
    * @return the result fo applied refactoring
    * @throws RefactoringException when there are no corresponding refactoring session
-   * @throws CoreException when impossible to apply rename refactoring
+   * @throws CoreException when impossible to apply recommend refactoring
    */
   @POST
-  @Path("rename/linked/apply")
+  @Path("recommend/linked/apply")
   @Consumes("application/json")
   @Produces("application/json")
   public RefactoringResult applyLinkedModeRename(LinkedRenameRefactoringApply refactoringApply)
       throws RefactoringException, CoreException {
-    return manager.applyLinkedRename(refactoringApply);
+
+    RefactoringResult res = manager.applyLinkedRename(refactoringApply);
+    JavaRenameRecommend javaRenameRecommend = new JavaRenameRecommend();
+    DataModel renameData = getInfo(refactoringApply);
+
+    if (renameData == null) JavaRenameRecommend.result.recommendOriginalName = "";
+    else javaRenameRecommend.recommend(renameData);
+    return res;
+  }
+
+  @POST
+  @Path("recommend/linked/extension/recommendation")
+  @Consumes("application/json")
+  @Produces("application/json")
+  public String getRecommendation() {
+    if (JavaRenameRecommend.result.recommendOriginalName.equals("")) return "$$null";
+    return JavaRenameRecommend.result.recommendRefactorType
+        + " "
+        + JavaRenameRecommend.result.recommendOriginalName
+        + " to "
+        + JavaRenameRecommend.result.recommendSubsequentName;
+  }
+
+  @POST
+  @Path("recommend/linked/extension/position")
+  @Consumes("application/json")
+  @Produces("application/json")
+  public String getRecommendationPosition() {
+    return ""
```

Comment:
are you sure it will return json?

Reply:
It was due to 'copy' and 'paste' and was my fault. Because my ip-validation is failing, I executed 'revert'. I am sorry about that. And this review make me seeing more problems in my code. I will correct them as soon as possible.